core/src/types.rs

```diff
@@ -10,6 +10,14 @@ use smallvec::{smallvec, SmallVec};
 #[serde(transparent)]
 pub struct PlayerID(pub usize);
 
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, Hash, Eq, PartialEq)]
+pub struct PlayerResult {
```

Comment:
Let's move this to `game_state.rs` and name it `PlayerGameFinishedResult` or something? It's not really a generic game-independent type.